tests_e2e/tests/agent_wait_for_cloud_init/agent_wait_for_cloud_init.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import time
+
+from assertpy import fail
+
+from tests_e2e.tests.lib.agent_test import AgentVmTest
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.shell import CommandError
+from tests_e2e.tests.lib.ssh_client import SshClient
+
+
+class AgentWaitForCloudInit(AgentVmTest):
+    """
+    This test verifies that the Agent waits for cloud-init to complete before it starts processing extensions.
+
+    To do this, it adds 'CloudInitScript' in cloud-init's custom data. The script ensures first that the Agent
```

Comment:
Is this script run by cloud-init? How is cloud-init's custom data used

Reply:
Yes, run by cloud-init. 

The configuration for cloud-init is passed (in Azure) in the custom data: https://learn.microsoft.com/en-us/azure/virtual-machines/linux/tutorial-automate-vm-deployment

There's quite a few things one can do with cloud-init. Here are some samples for Azure: https://learn.microsoft.com/en-us/azure/virtual-machines/linux/using-cloud-init

There is additional information in the readthedocs official documentation for cloud-init.

In the case of this test, it is executing a bash script